client/src/api/v2/camera.js

```diff
@@ -51,6 +51,59 @@ export function useOverlays(device) {
   return { overlays, setActiveOverlay };
 }
 
+/**
+ * @typedef {object} CameraMessageState
+ * @property {string} message The message currently entered by the user.
+ * @property {boolean} received Specifies whether a sent message has been
+ *                              acknowledged by the server in the last 5
+ *                              seconds.
+ * @property {boolean} sending Specifies whether we are waiting for an
+ *                             acknowledgement for the most recent message.
+ */
+
+/**
+ * @typedef {object} CameraMessageHook
+ * @property {CameraMessageState} state Current state of the messages.
+ * @property {function(string)} setMessage Set the message.
+ * @property {function(void)} sendMessage Send the message to the rider.
+ */
+
+/**
+ * Handle storing, sending and receiving acknowledgement of messages to the
+ * camera overlay.
+ *
+ * @returns {CameraMessageHook} Hook
+ */
+export function useMessageState() {
+  const [message, setMessage] = useState('');
+  const [received, setReceived] = useState(false);
+  const [sending, setSending] = useState(false);
+  const receivedTimeout = useRef(null);
+
+  const receivedCallback = () => {
+    clearTimeout(receivedTimeout.current);
+    setMessage('');
+    setReceived(true);
+    setSending(false);
+
+    // Remove the "received" status after 5 seconds
+    receivedTimeout.current = setTimeout(() => {
+      setReceived(false);
+    }, 5000);
+  };
+
+  useChannel('send-message-received', receivedCallback);
+
+  const sendMessage = () => {
+    if (!message) return;
+
+    emit('send-message', message);
+    setSending(true);
+  };
+
+  return { message, received, sending, setMessage, sendMessage };
```

Comment:
I think the return type is not exactly accurate as `CameraMessageHook` has a `state: CameraMessageState` property instead of `message: string`, `received: boolean`, and `sending: boolean`. May want to either remove `CameraMessageState` or update the return value.